compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -693,7 +694,7 @@ abstract class PrepJSInterop extends plugins.PluginComponent
         }
 
         val ownerLoadSpec = jsInterop.jsNativeLoadSpecOf(sym.owner)
-        val jsName = jsInterop.jsNameOf(sym)
+        val JSName.Literal(jsName) = jsInterop.jsNameOf(sym)
```

Comment:
Ah. That may well be the case, indeed. I hadn't though about that :-s

I suspect we don't need to change the IR for that, though. `js.constructor[Foo.A]` should translate to `JSBracketSelect(JSLoadModule("Foo$"), <refTo sym>)`